modules/engine/src/test/scala/observe/engine/StepSpec.scala

```diff
@@ -429,30 +425,31 @@ class StepSpec extends CatsSuite {
 
     val qss = q
       .flatMap { k =>
-        k.enqueue1(Event.start(seqId, user, clientId))
+        k.offer(Event.start(seqId, user, clientId))
           .flatMap(_ =>
             executionEngine
-              .process(PartialFunction.empty)(k.dequeue)(qs0)
+              .process(PartialFunction.empty)(Stream.fromQueueUnterminated(k))(qs0)
               .drop(1)
               .takeThrough(a => !isFinished(a._2.sequences(seqId).status))
               .compile
               .toVector
           )
       }
-      .unsafeRunSync()
 
-    val actionsCompleted = qss.map(_._1).collect { case SystemUpdate(x: Completed[_], _) => x }
-    assert(actionsCompleted.length == 4)
+    qss.map { x =>
```

Comment:
I think these are indepedent IO asserts right? If so you need to chain them and return them all at the end or onle the last one will be executed
Like `assert(actionsCompleted.length == 4) *> assert(executionsCompleted...)`
Also you may want to use `assertEquals`

Reply:
@cquiroz I may be misunderstanding something, or using the wrong asserts, but those come from munit. They all produce `Unit`, and handle test failures through side effects.